app/views/pages/developer-guide/modules/modules.liquid

```diff
@@ -1,16 +1,24 @@
 ---
 converter: markdown
 metadata:
-  title: Modules
+  title: Modules Introduction
   description: Modules allow code reuse and sharing while protecting the IP of creators.
   toc: true
 ---
 
-**Modules** allow code reuse and sharing, while protecting the IP of creators.
+**Modules** allow code reuse and sharing, while optionally protecting the IP of creators by using `private` folder. They can also be used as a namespace, to group various files (graphql, pages, partials etc.) related to a bigger functionality of a project, all in one directory.
+
+{% include 'alert/note', content: 'By default module files are not deleted during pos-cli deploy/sync to avoid breaking your application in a scenario when module contains private files to which you might not have access. You can change it by adding module name that you want to synchronize with your codebase to `modules_that_allow_delete_on_deploy` in [app/config.yml](/developer-guide/platformos-workflow/directory-structure/config). This behaviour is meant to be improved in the near future.' %}
```

Comment:
```suggestion
{% include 'alert/note', content: 'By default, module files are not deleted during `pos-cli deploy` or `sync` to avoid breaking your application in cases where a module contains private files to which you might not have access. You can change this by adding the module name you want to synchronize with your codebase to `modules_that_allow_delete_on_deploy` in [app/config.yml](/developer-guide/platformos-workflow/directory-structure/config). This behavior is expected to be improved in the near future.' %}
```